docs/sources/setup/migrate/ssd-to-distributed/_index.md

```diff
@@ -0,0 +1,372 @@
+
+# SSD
+backend:
+ replicas: 2
+read:
+ replicas: 3
+write:
+ replicas: 3
+
+# Distributed Loki
+ingester:
+ replicas: 0
+ zoneAwareReplication:
+  enabled: false
+
+querier:
+ replicas: 0
+ maxUnavailable: 0
+queryFrontend:
+ replicas: 0
+ maxUnavailable: 0
+queryScheduler:
+ replicas: 0
+distributor:
+ replicas: 0
+ maxUnavailable: 0
+compactor:
+ replicas: 0
+indexGateway:
+ replicas: 0
+ maxUnavailable: 0
+ruler:
+ replicas: 0
+ maxUnavailable: 0
+
+# Single binary Loki
+singleBinary:
+ replicas: 0
+
+minio:
+ enabled: false
+```
+
+{{< admonition type="note" >}}
+This example is only a reference on the parameters that need to be changed. There will be other parameters within your own config such as `limits_config`, `gateway`, `compactor`, etc. These can remain the same.
+{{< /admonition >}}
+
+## Stage 1: Deploying the Loki distributed components
+
+In this stage, we will deploy the distributed Loki components alongside the SSD components. We will also change the `deploymentMode` to `SimpleScalable<->Distributed`. The `SimpleScalable<->Distributed` migration mode allows for a zero-downtime transition between Simple Scalable and fully Distributed architectures. During migration, both deployment types run simultaneously, sharing the same object storage backend.
+
+```
+Simple Scalable Components   ->    Distributed Components
+------------------------    ->    ---------------------
+write (StatefulSet)         ->    Distributor + Ingester
+read (Deployment)           ->    Query Frontend + Querier
+backend (StatefulSet)       ->    Compactor + Ruler + Index Gateway
+```
+Request routing:
+
+The Gateway (nginx) handles request routing based on endpoint type:
+1. Write Path (`loki/api/v1/push`):
+   * Initially routes to Simple Scalable write component
+   * Gradually shifted to the Distributor
+   * Both write paths share the same object storage, ensuring data consistency
+2. Read Path (`/loki/api/v1/query`):
+   * Routes to either Simple Scalable read or Distributed Query Frontend
+   * Query results are consistent since both architectures read from same storage
+3. Admin/Background Operations:
+   * Compaction, retention, and rule evaluation handled by either backend or respective distributed components
+   * Operations are coordinated through object storage locks
+
+To start the migration process:  
+
+1. Create a copy of your existing `values.yaml` file and name it `values-migration.yaml` (You can also make these changes within your current values file).
+
+    ```bash
+    cp values.yaml values-migration.yaml
+    ```
+
+1. Next modify the following parameters; `deploymentMode`, `ingester` and components based on the annotations below.
+
+   ```yaml
+   ---
+   loki:
+     schemaConfig:
+       configs:
+         - from: "2024-04-01"
+           store: tsdb
+           object_store: s3
+           schema: v13
+           index:
+             prefix: loki_index_
+             period: 24h
+     storage_config:
+       aws:
```

Comment:
I don't think you've mentioned anywhere in the text that this example is specific for Amazon S3?  You might want to mention that if you're using Azure or GCS what you'd need to change in those files?  OR should it be very similar?